src/api/migraphx.py

```diff
@@ -163,8 +163,19 @@ def shapes(h):
              returns='const migraphx::shape&')
 
 
+@auto_handle
+def module(h):
+    h.constructor('create')
```

Comment:
I dont think we should allow constructing a module. A module should only come from the program(ie `get_main_module`), and the program will own the module, so there is no need to expose a `destroy` or `create` function either.

Reply:
So PR#673 adds the ability to not generate the `destroy` method. It just needs to be changed to `@auto_handle(ref=True)`.

Reply:
For the c++ api, looks like handle_base struct needs a destroy function, how can we handle that? definition of the handle base is 
template <class T, class D, D Deleter>
struct handle_base
{}